code/game/machinery/doors/airlock_electronics.dm

```diff
@@ -17,7 +17,7 @@
 	var/autoset = TRUE // Whether the door should inherit access from surrounding areas
 
 /obj/item/airlock_electronics/attack_self(mob/user)
-	if(!ishuman(user) || !issilicon(user))
+	if (!ishuman(user) && !istype(user,/mob/living/silicon/robot))
```

Comment:
```suggestion
	if(!ishuman(user) || !isrobot(user)
```

We have macros for this for a reason ahh. Also you probably meant or. Try using it as a monkey on your current code and it'd probably let you.

Reply:
the && is the bugfix so using || completely negates it. also checking for robot instead of silicon is arbitrary, just sue issilicon

Reply:
Hold on a bit